src/components/navbutton.js

```diff
@@ -0,0 +1,91 @@
+/** @jsx jsx */
+import { Box, Flex, NavLink, Image, jsx } from 'theme-ui'
+import { Component } from 'react'
+import React from 'react'
+import { BsCaretDownFill, BsCaretUpFill } from 'react-icons/bs'
+
+class NavButton extends Component {
+  constructor(props) {
+    super(props);
+    const {loc,text,icon,active,dropdown} = props;
+    this.loc = loc;
+    this.text = text;
+    this.active = active;
+    this.icon = icon;
+    this.dropdown = dropdown;
+    this.state = {open:false};
+    this.expand = ()=>this.setState({open:true});
+    this.collapse = ()=>this.setState({open:false});
+  }
+  render() {
+    return (
+      <Box
+        sx={{
+          padding: '4px 10px',
+          margin: '0 auto',
+          borderRadius: this.state.open?'10px 10px 0 0':'10px',
+          background: this.active||this.state.open?'#dfd':'inherit'
+        }}
+        onBlur={this.collapse}
+      >
+        <Flex
+          sx={{
+            alignItems:'center',
+            justifyContent:'center'
+          }}
+        >
+          <NavLink
+            href={this.loc}
+          >
+            {this.text}
+            {this.icon&&jsx(this.icon,{
```

Comment:
Do not use element factory directly, use JSX syntax (and name the prop accordingly)